shm/observer.hpp

```diff
@@ -0,0 +1,5 @@
+class Observer {
+public:
+    virtual void nextDay()=0;
+    //virtual ~Observer();
```

Comment:
Remove comment. Observer has virtual method, so destructor also should be virtual.